### DISCUSSION TITLE
Complex modulus in Homogenization kernel

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#23525] [Q&A Modules: Solid mechanics]
I am trying to calculate the homogenized properties for a periodic unit cell with complex valued youngs modulus. I tried creating 4 different stress divergence kernels and passing them to the homogenization kernels, but that doesn't seem to work. Has anyone tried to do this or or could guide me on how to do this?

My Input File
[2D_homogenization_complex_modulus.txt](https://github.com/idaholab/moose/files/10821042/2D_homogenization_complex_modulus.txt)

**Comment by GiudGiud** [comment]
@jiangwen84 @dschwen

**Comment by hugary1995** [comment]
In general, you need to use different base names for the real and imaginary parts. That's all I can say without seeing the equations you want to solve.
